File: code/js/src/Domain/tokens/AccessToken.ts
import { AccessTokenOutputModel } from '../../Dto/output/credentials/AccessTokenOutputModel';

/**
 * Represents an access token.
 *
 * An access token is used to authenticate a user in the system.
 *
 * @param token The token.
 * @param expiresAt The date and time when the token expires.
 */
export class AccessToken {
    constructor(
        readonly token: string,
        readonly expiresAt: Date,
    ) {}

    static fromDto(dto: AccessTokenOutputModel): AccessToken {
        return new AccessToken(dto.token, new Date(dto.expiresAt));
    }

    isExpired(): boolean {
        return this.expiresAt < new Date();
    }
}

File: code/server/jvm/im-domain/src/main/kotlin/im/domain/tokens/AccessToken.kt
package im.domain.tokens

import im.domain.sessions.Session
import java.time.LocalDateTime
import java.util.UUID

/**
 * An access token that can be used to authenticate requests.
 *
 * @property token The unique identifier of the access token.
 * @property session The session associated with the access token.
 * @property expiresAt The date and time when the access token expires.
 * @property expired Indicates if the access token has expired.
 */
data class AccessToken(
    val token: UUID = UUID.randomUUID(),
    val session: Session,
    val expiresAt: LocalDateTime,
) {
    val expired: Boolean
        get() = expiresAt.isBefore(LocalDateTime.now())
}
